lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.ts

```diff
@@ -16,66 +16,85 @@
-    const select = new Subject<Node[]>();
-    select.subscribe({
-      complete: this.close.bind(this)
-    });
-    const data = <ContentNodeSelectorComponentData> {
-      title: 'Select a file',
-      actionName: NodeAction.ATTACH,
-      currentFolderId,
-      restrictRootToCurrentFolderId,
-      select,
-      selectionMode,
-      isSelectionValid: (entry: Node) => entry.isFile,
-      showFilesInResult: true,
-      showDropdownSiteList: false,
-      showLocalUploadButton: isAllFileSources
-  };
-    this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
-    return select;
-  }
+    constructor(
+        private apiService: AlfrescoApiService,
+        private logService: LogService,
+        private notificationService: NotificationService,
+        private dialog: MatDialog) {
+    }
+
+    openUploadFileDialog(currentFolderId?: string, selectionMode?: string, isAllFileSources?: boolean, restrictRootToCurrentFolderId?: boolean): Observable<Node[]> {
+        const select = new Subject<Node[]>();
+        select.subscribe({
+        complete: this.close.bind(this)
+        });
+        const data = <ContentNodeSelectorComponentData> {
+        title: 'Select a file',
+        actionName: NodeAction.ATTACH,
+        currentFolderId,
+        restrictRootToCurrentFolderId,
+        select,
+        selectionMode,
+        isSelectionValid: (entry: Node) => entry.isFile,
+        showFilesInResult: true,
+        showDropdownSiteList: false,
+        showLocalUploadButton: isAllFileSources
+    };
+        this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
+        return select;
+    }
 
-    async fetchNodeIdFromRelativePath(alias: string, opts: { relativePath: string }): Promise<string> {
-        const relativePathNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias, opts)
-        .catch((err) => this.handleError(err));
-        return relativePathNodeEntry?.entry?.id;
+    fetchNodeFromRelativePath(alias: string, opts: { relativePath: string }): Promise<NodeEntry> {
+        return from(this.apiService.nodesApi.getNode(alias, opts)).pipe(
+            catchError((error) => {
+                this.sourceNodeNotFound = true;
+                this.handleError(error);
+                return of(<NodeEntry> {});
+            })
+        ).toPromise();
     }
 
-    async fetchAliasNodeId(alias: string): Promise<string> {
-        const aliasNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias)
-        .catch((err) => this.handleError(err));
-        return aliasNodeEntry?.entry?.id;
+    fetchAliasNode(alias: string): Promise<NodeEntry> {
+        return from(this.apiService.nodesApi.getNode(alias)).pipe(
+            catchError((error) => {
+                this.handleError(error);
+                return of(<NodeEntry> {});
```

Comment:
typecasting an empty object to avoid compilation errors I am not sure is the right thing to do. Can we investigate what is a good practice for that and what do we do in other places?

Reply:
rollbacked